invenio_rdm_records/requests/access/requests.py

```diff
@@ -0,0 +1,215 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 TU Wien.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Access requests for records."""
+
+from flask import current_app, g
+from invenio_access.permissions import authenticated_user, system_identity
+from invenio_i18n import lazy_gettext as _
+from invenio_mail.tasks import send_email
+from invenio_records_resources.services.uow import Operation, RecordCommitOp
+from invenio_requests.customizations import RequestType, actions
+from invenio_requests.customizations.event_types import CommentEventType
+from invenio_requests.proxies import current_events_service as events_service
+from marshmallow import fields
+
+from ...proxies import current_rdm_records_service as service
+
+
+class EmailOp(Operation):
+    """A notification operation."""
+
+    def __init__(self, receiver, subject, html_body, body):
+        """Initialize operation."""
+        self.receiver = receiver
+        self.subject = subject
+        self.html_body = html_body
+        self.body = body
+
+    def on_post_commit(self, uow):
+        """Start task to send text via email."""
+        send_email(
+            {
+                "subject": self.subject,
+                "html_body": self.html_body,
+                "body": self.body,
+                "recipients": [self.receiver],
+                "sender": current_app.config["MAIL_DEFAULT_SENDER"],
+            }
+        )
+
+
+#
+# Actions
+#
+class UserSubmitAction(actions.SubmitAction):
```

Comment:
Question: Should we notify via email that an user/guest requested access?

Reply:
Yes, I think email notifications are a massive improvement to the usefulness of requests.
It's one of the things that I'm gonna create follow-up issues for.